http-clients/netty-nio-client/src/main/java/software/amazon/awssdk/http/nio/netty/internal/utils/NettyUtils.java

```diff
@@ -132,9 +135,29 @@ public static void doInEventLoop(EventExecutor eventExecutor, Runnable runnable)
      */
     public static void doInEventLoop(EventExecutor eventExecutor, Runnable runnable, Promise<?> promise) {
         try {
-            doInEventLoop(eventExecutor, runnable);
+            if (eventExecutor.inEventLoop()) {
+                runnable.run();
+            } else {
+                eventExecutor.submit(() -> {
+                    try {
+                        runnable.run();
+                    } catch (Exception e) {
+                        promise.setFailure(e);
```

Comment:
should we just change this to `Throwable`?

Reply:
Updated to Throwable everywhere in this class